firefox-ios/Client/Nimbus/OnboardingFeatureLayer/NimbusOnboardingKitFeatureLayer.swift

```diff
@@ -71,6 +78,20 @@ class NimbusOnboardingKitFeatureLayer: NimbusOnboardingFeatureLayerProtocol {
                 imageID: card.imageID,
                 instructionsPopup: card.instructionsPopup)
         }
+        // Filter out cards that are not relevant for the current device type and browser state.
+        .filter { viewModel in
+            // Filter out cards that are not relevant for the current device type.
+            if isIpad, let action = viewModel.multipleChoiceButtons.first?.action,
+               action == .toolbarTop || action == .toolbarBottom {
+                return false
+            }
+
+            // Filter out welcome cards for DMA users who already have Firefox set as default browser
+            if isDefaultBrowser && viewModel.name.localizedCaseInsensitiveContains("welcome") {
+                return false
+            }
+            return true
```

Comment:
this seems a bit brittle if we ever change the content. Could we filter based on card type instead?

Reply:
Looks like a great candidate but I haven't seen it used. There's also:
```
open-ios-fx-settings:
        description: >
          Will take the user to the default browser settings
          in the iOS system settings
```
Which is a good candidate.